src/contracts/pods/EigenPod.sol

```diff
@@ -159,143 +113,141 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
+    function verifyCheckpointProofs(
+        BeaconChainProofs.BalanceContainerProof calldata balanceContainerProof,
+        BeaconChainProofs.BalanceProof[] calldata proofs
+    ) 
+        external 
+        onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_CHECKPOINT_PROOFS) 
+    {
+        uint64 beaconTimestamp = currentCheckpointTimestamp;
         require(
-            (validatorFields.length == validatorFieldsProofs.length) &&
-                (validatorFieldsProofs.length == withdrawalProofs.length) &&
-                (withdrawalProofs.length == withdrawalFields.length),
-            "EigenPod.verifyAndProcessWithdrawals: inputs must be same length"
+            beaconTimestamp != 0, 
+            "EigenPod.verifyCheckpointProofs: must have active checkpoint to perform checkpoint proof"
         );
 
-        // Verify passed-in beaconStateRoot against oracle-provided block root:
-        BeaconChainProofs.verifyStateRootAgainstLatestBlockRoot({
-            latestBlockRoot: eigenPodManager.getBlockRootAtTimestamp(oracleTimestamp),
-            beaconStateRoot: stateRootProof.beaconStateRoot,
-            stateRootProof: stateRootProof.proof
+        Checkpoint memory checkpoint = _currentCheckpoint;
+
+        // Verify `balanceContainerProof` against `beaconBlockRoot`
+        BeaconChainProofs.verifyBalanceContainer({
+            beaconBlockRoot: checkpoint.beaconBlockRoot,
+            proof: balanceContainerProof
         });
 
-        VerifiedWithdrawal memory withdrawalSummary;
-        for (uint256 i = 0; i < withdrawalFields.length; i++) {
-            VerifiedWithdrawal memory verifiedWithdrawal = _verifyAndProcessWithdrawal(
-                stateRootProof.beaconStateRoot,
-                withdrawalProofs[i],
-                validatorFieldsProofs[i],
-                validatorFields[i],
-                withdrawalFields[i]
-            );
+        // Process each checkpoint proof submitted
+        uint64 exitedBalancesGwei;
+        for (uint256 i = 0; i < proofs.length; i++) {
+            ValidatorInfo memory validatorInfo = _validatorPubkeyHashToInfo[proofs[i].pubkeyHash];
+
+            // Validator must be in the ACTIVE state to be provable during a checkpoint.
+            // Validators become ACTIVE when initially proven via verifyWithdrawalCredentials
+            // Validators become WITHDRAWN when a checkpoint proof shows they have 0 balance
+            if (validatorInfo.status != VALIDATOR_STATUS.ACTIVE) {
+                continue;
+            }
+            
+            // Ensure we aren't proving a validator twice for the same checkpoint. This will fail if:
+            // - validator submitted twice during this checkpoint
+            // - validator withdrawal credentials verified after checkpoint starts, then submitted
+            //   as a checkpoint proof
+            if (validatorInfo.mostRecentBalanceUpdateTimestamp >= beaconTimestamp) {
+                continue;
+            }
```

Comment:
Bug: this condition prevents the following flow -
1. verifyWC + startCheckpoint in the same block
3. completeCheckpoint

Reply:
Fixed in [6f9da7f](https://github.com/Layr-Labs/eigenlayer-contracts/pull/515/commits/6f9da7f9e7b6358eb2cf5307cf88c2df802aad88)